ngsderive/utils.py

```diff
@@ -68,18 +69,23 @@ def __next__(self):
             if self.gzipped:
                 query_name = query_name.decode("utf-8")
                 query = query.decode("utf-8")
+                quality = quality.decode("utf-8")
```

Comment:
I might conditionally decode this is store qualities is set to true. Else you do that work even if the quality will never  be used.

Reply:
This will very likely get you much more performance benefit that what the conditional below would do (simply exclude it from the yielded dictionary).